Core/Src/main.c

```diff
@@ -87,16 +89,37 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
+  /*CS pull up to end any ongoing transmissions*/
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
+  __HAL_TIM_SET_COMPARE(&htim1, TIM_CHANNEL_1, 0.05 * 64000); /*set compare register value to 5% duty cycle time*/
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1); /*start the timer*/
+  /* takes in 3 bytes of data format found in figures 6-1 and 6-2*/
+  uint8_t data_transmit[3] = {1, 128, 0}; /*This connects to channel 0 where D2 D1 D0 are x00 respectively*/
```

Comment:
usually when we are working with the number that is address relevant or byte relevant, we use hexadecimal representation on the numbers. for example, you can use `0x80` instead of 128 here.

Reply:
And also, one more suggestion for good practice, you could add a `const` keyword to this tx buffer since it will never be changed in this bootcamp.